tools/pr-docs

```diff
@@ -0,0 +1,34 @@
+#!/bin/bash -e
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+if [ "$#" -ne 2 ] ; then
+    echo "Usage: $0 pr-number output/location" >&2
+    echo "Render documentation from a pull request" >&2
+    exit 1
+fi
+
+pr=$1
+outdir=$2
+
+# Linux || OSX
+tmpdir=$(mktemp -d --tmpdir 2>/dev/null || mkdir -d -t spectre-doc)
+trap 'rm -rf "${tmpdir}"' EXIT
+
+spectre=$(git rev-parse --show-toplevel)
+
+pushd "${tmpdir}" >/dev/null
+# Clone from the local SpECTRE repo to avoid having to redownload
+# everything.
+git clone --shared --quiet --no-checkout -- "${spectre}" source
+pushd source >/dev/null
+git fetch https://github.com/sxs-collaboration/spectre pull/"${pr}"/head
+git checkout --quiet FETCH_HEAD
+popd >/dev/null
+cmake -D DOCS_ONLY=TRUE source
```

Comment:
If I run this in a new directory that has never built anything it fails inside SetupCharm. Is this intentional? If so could you add a comment saying something about this?